src/cryptonote_core/blockchain.cpp

```diff
@@ -441,6 +441,55 @@ bool Blockchain::init(BlockchainDB* db, const network_type nettype, bool offline
   MINFO("Blockchain initialized. last block: " << m_db->height() - 1 << ", " << epee::misc_utils::get_time_interval_string(timestamp_diff) << " time ago, current difficulty: " << get_difficulty_for_next_block());
   m_db->block_txn_stop();
 
+  uint64_t num_popped_blocks = 0;
+  while (true)
+  {
+    uint64_t top_height;
+    crypto::hash top_id;
+    top_id = get_tail_id(top_height);
+    bool orphan;
+    block top_block;
+    get_block_by_hash(top_id, top_block, &orphan);
```

Comment:
That's really m_db->get_block, since you don't want to get an orphan here, but something from the chain. It shouldn't happen since syncing's not yet enabled so you won't get other threads modifying the db but it still seems safer.

Reply:
OK, I use `m_db->height()` and `m_db->top_block_hash()` and `m_db->get_top_block()` now.